test/unittest/statistics/rtps/RTPSStatisticsTests.cpp

```diff
@@ -0,0 +1,225 @@
+        //   ... unless it's associated to other entity
+        EXPECT_TRUE(participant->add_statistics_listener(listener1, another_kind));
+        // + fails if an unknown listener is removed
+        EXPECT_FALSE(participant->remove_statistics_listener(listener2, kind));
+        // + fails if a known listener is removed with a non registered entity
+        EXPECT_FALSE(participant->remove_statistics_listener(listener1, yet_another_kind));
+        // + succeeds to remove a known listener
+        EXPECT_TRUE(participant->remove_statistics_listener(listener1, kind));
+        // + fails if a listener is already removed
+        EXPECT_FALSE(participant->remove_statistics_listener(listener1, kind));
+
+        // test the writer apis
+        // + fails if no listener has been yet added
+        EXPECT_FALSE(writer->remove_statistics_listener(listener1));
+        // + fails to add an empty listener
+        EXPECT_FALSE(writer->add_statistics_listener(nolistener));
+        // + succeeds to add a new listener
+        ASSERT_TRUE(writer->add_statistics_listener(listener1));
+        // + fails to add multiple times the same listener
+        EXPECT_FALSE(writer->add_statistics_listener(listener1));
+        // + fails if an unknown listener is removed
+        EXPECT_FALSE(writer->remove_statistics_listener(listener2));
+        // + succeeds to remove a known listener
+        EXPECT_TRUE(writer->remove_statistics_listener(listener1));
+        // + fails if a listener is already removed
+        EXPECT_FALSE(writer->remove_statistics_listener(listener1));
+
+        // test the reader apis
+        // + fails if no listener has been yet added
+        EXPECT_FALSE(reader->remove_statistics_listener(listener1));
+        // + fails to add an empty listener
+        EXPECT_FALSE(reader->add_statistics_listener(nolistener));
+        // + succeeds to add a new listener
+        ASSERT_TRUE(reader->add_statistics_listener(listener1));
+        // + fails to add multiple times the same listener
+        EXPECT_FALSE(reader->add_statistics_listener(listener1));
+        // + fails if an unknown listener is removed
+        EXPECT_FALSE(reader->remove_statistics_listener(listener2));
+        // + succeeds to remove a known listener
+        EXPECT_TRUE(reader->remove_statistics_listener(listener1));
+        // + fails if a listener is already removed
+        EXPECT_FALSE(reader->remove_statistics_listener(listener1));
+    }
+
+    // Check PUBLICATION_THROUGHPUT and SUBSCRIPTION_THROUGHPUT callbacks are performed
+    {
+        using namespace ::testing;
+
+        auto participant_listener = make_shared<MockListener>();
+        auto writer_listener = make_shared<MockListener>();
+        auto reader_listener = make_shared<MockListener>();
+        ASSERT_TRUE(participant->add_statistics_listener(participant_listener,
+                fastdds::statistics::EventKind::DISCOVERED_ENTITY));
```

Comment:
This listener is not removed later. `RTPSDomain::removeRTPSParticipant(participant)` may fail.

Reply:
Nope, by policy dangling resources don't lead to an error on endpoint destruction.